File: techila/examples/Matlab/Tutorial/4_nestedloops/local_function.m
function result = local_function()
% This function can be executed locally on the End-Users computer. The
% function multiplies each element in the matrix A = [2 4 6; 8 10 16; 14 16 18] 
% individually.
%
% To use execute the function locally, use command:
%
% result = local_function()

% Copyright 2010-2013 Techila Technologies Ltd.

A = [2 4 6; 8 10 16; 14 16 18]; % Create matrix A
Imax=size(A,1); % Number of rows
Jmax=size(A,2); % Number of columns

for i = 1:Imax
    for j = 1:Jmax
        result(i,j) = A(i,j)*2; % Multiply each element individually
    end
end

File: techila/examples/R/cloudfor/2_transferring_data_files/local_function.r
# Copyright 2012-2013 Techila Technologies Ltd.

local_function <- function() {
  # This function will be executed locally on your computer and will not
  # communicate with the Techila environment.
  #
  # Usage:
  #
  # result <- local_function()

  # Read values from the data files
  values <- read.table("datafile.txt", header=TRUE, as.is=TRUE)
  targetvalue <- read.table("datafile2.txt", header=TRUE, as.is=TRUE)

  # Determine the number of rows and columns of data
  rows <- nrow(values)
  cols <- ncol(values)

  # Create empty matrix for results
  result <- matrix(rep(NA, 12), rows, cols)

  for (i in 1:rows) { # For each row of data

    data <- values[i,] # Read the values on the row
    for (j in 1:cols) { # For each element on the row

      # Compare values on the row to the ones on the target row
      if(identical(values[[i, j]], targetvalue[[j]])) {
        result[i, j] <- TRUE  # If rows match
      }
      else {
        result[i, j] <- FALSE # If rows don't match
      }
    }
  }
  print(result)
  result
}


